content/en/blog/_posts/2021-07-16-read-write-once-pod-access-mode-alpha.md

```diff
@@ -0,0 +1,269 @@
+
+If you create another pod that references the same PVC with this access mode, the pod will fail to start like so:
+
+```
+Events:
+  Type     Reason            Age   From               Message
+  ----     ------            ----  ----               -------
+  Warning  FailedScheduling  1s    default-scheduler  0/1 nodes are available: 1 node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode.
+```
+
+### How is this different than the ReadWriteOnce access mode?
+
+The ReadWriteOnce access mode restricts volume access to a single *node*, which means it is possible for multiple pods on the same node to read from and write to the same volume.
+
+Could this cause problems?
+Potentially yes!
+If you are running workloads that access sensitive data, other pods on that node can access it (if updated to mount the volume).
+Unfortunately, the only way to work around this is by running a single pod on the node, which is an inefficient use of compute resources in your cluster.
+
+With ReadWriteOncePod these restrictions go away.
+You don't have to worry about neighboring pods potentially accessing your volumes.
+Just set the access mode on your PVC, and Kubernetes will guarantee only a single pod has access.
+
+## How do I use it?
+
+The ReadWriteOncePod access mode is in alpha for Kubernetes 1.22.
+As a first step you need to enable the ReadWriteOncePod [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates) for `kube-apiserver`, `kube-scheduler`, and `kubelet`.
+You can enable (or disable) the feature by setting command line arguments:
+
+```
+--feature-gates="...,ReadWriteOncePod=true"
+```
+
+[here]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+
+### Creating a PersistentVolumeClaim
+
+In order to use the ReadWriteOncePod access mode for your PVs and PVCs, you will need to create a new PVC with the access mode:
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: top-secret
+spec:
+  accessModes:
+  - ReadWriteOncePod # Allow only a single pod to access top-secret.
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+If your storage plugin supports [dynamic provisioning](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/), new PersistentVolumes will be created with the ReadWriteOncePod access mode applied.
+
+#### Migrating existing PersistentVolumes
+
+If you have existing PersistentVolumes, they can be migrated to use ReadWriteOncePod.
+
+As a prerequisite, you need to stop any workloads that are using the PersistentVolumeClaim bound to the PersistentVolume you want to migrate, and then delete the PersistentVolumeClaim:
+
+```
+kubectl delete pvc cat-pictures-pvc
+```
+
+Next you need to replace the PersistentVolume's access modes with ReadWriteOncePod:
+
+```yaml
+kind: PersistentVolume
```

Comment:
You will probably want to set PVC.storageClassName to "", otherwise it will use the default storageclass.

Or set the PV.StorageClassName to the same storageClassName that the original PVC was.